extensions/shared/components/audio-player/index.jsx

```diff
@@ -0,0 +1,160 @@
+/* global _wpmejsSettings, MediaElementPlayer */
+
+/**
+ * External dependencies
+ */
+import { throttle } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { STATE_PLAYING, STATE_PAUSED, STATE_ERROR } from './constants';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const meJsSettings = typeof _wpmejsSettings !== 'undefined' ? _wpmejsSettings : {};
+
+function createJumpButton( containerClass, label, clickHandler ) {
+	const buttonContainer = document.createElement( 'div' );
+	buttonContainer.className = containerClass;
+
+	const button = document.createElement( 'button' );
+	button.innerText = label;
+	button.addEventListener( 'click', clickHandler );
+	buttonContainer.appendChild( button );
+	return buttonContainer;
+}
+
+function AudioPlayer( {
+	trackSource,
+	onPlay,
+	onPause,
+	onError,
+	onTimeChange,
+	onSkipForward,
+	onJumpBack,
+	currentTime,
+	playStatus = STATE_PAUSED,
+} ) {
+	const audioRef = useRef();
+
+	/**
+	 * Play current audio.
+	 *
+	 * @public
+	 */
+	const play = () => {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		audioRef.current.play().catch( () => {} );
+	};
+
+	/**
+	 * Pause current audio.
+	 *
+	 * @public
+	 */
+	const pause = () => {
+		audioRef.current.pause();
```

Comment:
similar to above, but I think should do a quick return. 
```suggestion
		if ( ! audioRef?.current ) {
			return;
		}
		audioRef.current.pause();
```